### PR TITLE
perf: add js-web-frameworks benchmark

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

**Review by IgorMinar — CHANGES_REQUESTED** [review]
I pushed a commit with a few fixes + added a bunch of comments. I find the structure of the app to be hard to follow, but I don't understand what constraints you had to write it this way. I suggest separating code into 4 parts:
- the bootstrap code
- the component under test
- the code that trigers various actions
- the benchpress spec

File: modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+load("//modules/benchmarks:benchmark_test.bzl", "benchmark_test")
+
+package(default_visibility = ["//modules/benchmarks:__subpackages__"])
+
+ng_module(
+    name = "ng2",
+    srcs = glob(["*.ts"]),
+    generate_ve_shims = True,
+    tsconfig = "//modules/benchmarks:tsconfig-build.json",
+    # TODO: FW-1004 Type checking is currently not complete.
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel]: Yeh, copy & paste fail, removed...

File: modules/benchmarks/src/js-web-frameworks/ng2/init.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import {JsWebFrameworksComponent, JsWebFrameworksModule, RowData} from './rows';
+
+
+function _random(max: number) {
+  return Math.round(Math.random() * 1000) % max;
+}
+
+// TODO(pk): we probably need to support appending rows, but this would be hard(?) to test with
+// benchpress
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/js-web-frameworks/ng2/init.ts]: This was a left-over comment from my earlier experiments, removed. Thnx for catching!

File: modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    deps = [
+        "//modules/benchmarks/src:util_lib",
+        "//packages/core",
+        "//packages/platform-browser",
+    ],
+)
+
+ng_rollup_bundle(
+    name = "bundle",
+    entry_point = ":index_aot.ts",
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel]: Yeh, was trying to keep it in-line with all the other existing benchmarks but I think that we should have just `index.ts`. Renamed it here and opened a follow-up issues to cleanup other benchmarks.

File: modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,41 @@
+load("//tools:defaults.bzl", "ng_module", "ng_rollup_bundle", "ts_devserver")
+load("//modules/benchmarks:benchmark_test.bzl", "benchmark_test")
+
+package(default_visibility = ["//modules/benchmarks:__subpackages__"])
+
+ng_module(
+    name = "ng2",
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel]: Nope, most definitively not! The thing is that I did this benchmark so it follows the folder / file structure of all the other benchmarks (otherwise it would be confusing because of inconstancy). 

For a long time I wanted to do a bigger cleanup of the existing benchmarks but "there was never time for it" :-/ I've opened https://github.com/angular/angular/issues/34127 listing all the cleanups I think we should do.

We've got 2 options here: 

1. create this benchmark in the way we want to see the final structure, live with the inconsistency for a while, and then cleanup other benchmarks to match this one;
2. create this benchmark in the way that it follows structure of the existing benchmarks and then cleanup all benchmarks based on #34127

Personally I would prefer (2) but not feeling too strongly about it!

File: modules/benchmarks/src/js-web-frameworks/ng2/rows.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    }
+    this._appRef.tick();
+  }
+}
+
+@NgModule({
+  imports: [BrowserModule],
+  declarations: [JsWebFrameworksComponent],
+  bootstrap: [JsWebFrameworksComponent],
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply modules/benchmarks/src/js-web-frameworks/ng2/rows.ts]: Totally, couldn't agree more! Nope, most definitively not! The thing is that I did this benchmark so it follows the folder / file structure of all the other benchmarks (otherwise it would be confusing because of inconstancy). 

For a long time I wanted to do a bigger cleanup of the existing benchmarks but "there was never time for it" :-/ I've opened https://github.com/angular/angular/issues/34127 listing all the cleanups I think we should do.

We've got 2 options here: 

1. create this benchmark in the way we want to see the final structure, live with the inconsistency for a while, and then cleanup other benchmarks to match this one;
2. create this benchmark in the way that it follows structure of the existing benchmarks and then cleanup all benchmarks based on #34127

Personally I would prefer (2) but not feeling too strongly about it!

**Comment by pkozlowski-opensource** [comment]
Thnx for the review @IgorMinar ! I've addressed smaller issues, but I guess the most important item is:

> I find the structure of the app to be hard to follow, but I don't understand what constraints you had to write it this way.

Totally with you! The only "constraint" I've imposed here was "consistency with the existing benchmarks folders / files structure".

I've opened #34127 so we can clean-up things. My personal preference would be to merge this PR so it matches the existing structure and then do across-the-board cleanup as described in #34127.
But we can also make this one "role model" and then cleanup other benchmarks.

WDYT?

In any case we need to coordinate any folder / file moves with the LL setup, if I'm not mistaken?

File: modules/benchmarks/src/js-web-frameworks/ng2/rows.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (IgorMinar) [reply modules/benchmarks/src/js-web-frameworks/ng2/rows.ts]: Let's go with #2 for now. Thanks for opening the issue.

File: modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel (hunk 4 above)
[the remark this answers is not in the document]
Comment (IgorMinar) [reply modules/benchmarks/src/js-web-frameworks/ng2/BUILD.bazel]: Answered below. Let's go with #2.

**Review by IgorMinar — APPROVED** [review]
Lgtm. Let's refactor the folder structure separately from this change as it will affect LL.

**Comment by IgorMinar** [comment]
Merge-assistance: no g3 impact. This is just a new benchmark
